free/src/main/scala/cats/free/Free.scala

```diff
@@ -250,3 +250,53 @@ object Free {
       def flatMap[A, B](a: Free[S, A])(f: A => Free[S, B]): Free[S, B] = a.flatMap(f)
     }
 }
+
+private trait FreeFoldable[F[_]] extends Foldable[Free[F, ?]] {
+
+  implicit def FoldableF: Foldable[F]
+  implicit def FunctorF: Functor[F]
+
+  override final def foldLeft[A,B](fa: Free[F, A], b: B)(f: (B, A) => B): B =
+    fa.fold(f(b,_), FoldableF.foldLeft(_,b)((bb,freeA) => foldLeft(freeA,bb)(f)))
+
+  override final def foldRight[A,B](fa: Free[F, A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+    fa.fold(f(_, lb), FoldableF.foldRight(_,lb)( (freeA,eb) => foldRight(freeA,eb)(f)))
+}
+
+private trait FreeTraverse[F[_]] extends Traverse[Free[F, ?]] with FreeFoldable[F] {
+  implicit def TraversableF: Traverse[F]
+
+  def FoldableF = TraversableF
+
+  override final def traverse[G[_], A, B](fa: Free[F, A])(f: A => G[B])(implicit G: Applicative[G]): G[Free[F, B]] =
+    fa.fold(
+      a => G.map(f(a))(Free.pure(_)),
+      ffreeA => G.map(TraversableF.traverse(ffreeA)(traverse(_)(f)))(Free.roll(_))
+    )
+
+  // Override Traverse's map to use Free's map for better performance
+  override final def map[A,B](fa: Free[F, A])(f: A => B): Free[F, B] = fa.map(f)
+}
+
+sealed private[free] abstract class FreeInstances {
+
+  implicit def catsFoldableForFree[F[_]](
```

Comment:
It is not really documented outside issue https://github.com/typelevel/cats/issues/1061 itself (it should probably be added to [these guidelines](https://typelevel.org/cats/guidelines.html)): the implicit values in Cats are prefixed with the package.

So these implicits would be `catsFreeFoldableForFree` and `catsFreeTraverseForFree`.